src/server.js

```diff
@@ -671,14 +691,19 @@ if (cluster.isMaster && !module.parent) {
     return ensureKeystore(() => {
       if (ports.httpPort || ports.httpsPort) {
         koaMiddleware.setupApp((app) => {
-          if (ports.httpPort) { promises.push(startHttpServer(ports.httpPort, bindAddress, app)) }
+          if (ports.httpPort) {
+            promises.push(startHttpServer(ports.httpPort, bindAddress, app))
+          }
+
           if (ports.httpsPort) { promises.push(startHttpsServer(ports.httpsPort, bindAddress, app)) }
           return promises
         })
       }
 
       if (ports.apiPort && config.api.enabled) {
-        koaApi.setupApp(app => promises.push(startApiServer(ports.apiPort, bindAddress, app)))
+        config.api.protocol === 'https'
+          ? koaApi.setupApp(app => promises.push(startApiHttpsServer(ports.apiPort, bindAddress, app)))
+          : koaApi.setupApp(app => promises.push(startApiHttpServer(ports.apiPort, bindAddress, app)))
```

Comment:
@de-laz one more thing i thought about that could be an issue but haven't tested yet is that if a client upgrade to the new version with the additional config option but they are still making use of an old config file without this property, the server will then default to non-secure (http). 

Im wondering if this shouldn't rather be the other way round to ensure backward compatibility. So if they haven't supplied the protocol property then it defaults to `https` as it would have been in the previous version

Reply:
@BMartinos I agree with this, it makes sense.